src/rustdoc-json-types/lib.rs

```diff
@@ -565,6 +565,12 @@ pub enum Type {
         type_: Box<Type>,
         len: String,
     },
+    /// u32 is 0..=100
+    Pat {
```

Comment:
Could you bump the `FORMAT_VERSION` constant in the top of this file please.

Also, CI will fail unless you add a new case to

https://github.com/rust-lang/rust/blob/226b2496fc3ef997fa1cccf84259539203f82375/src/tools/jsondoclint/src/validator.rs#L243-L245

All it needs to do is recurse into the inner type.